dScripts/LegoDieRoll.cpp

```diff
@@ -0,0 +1,55 @@
+#include "LegoDieRoll.h"
+#include "Entity.h"
+#include "dLogger.h"
+#include "GameMessages.h"
+
+void LegoDieRoll::OnStartup(Entity* self) {
+    self->AddTimer("DoneRolling", 10.0f);
+    self->AddTimer("ThrowDice", LegoDieRoll::animTime);
+}
+
+void LegoDieRoll::OnTimerDone(Entity* self, std::string timerName) {
+    if (timerName == "DoneRolling") {
+        self->Smash(self->GetObjectID(), SILENT);
+    } 
+    else if (timerName == "ThrowDice") {
+        int dieRoll = GeneralUtils::GenerateRandomNumber<int>(1, 6);
+
+        switch (dieRoll)
+        {
+        case 1:
+		    GameMessages::SendPlayAnimation(self, u"roll-die-1");
```

Comment:
and just add an if statement for if it lands on 6

Reply:
I was thinking of that too, but I'm thinking this is better form. It's actually cleaner code since we don't need a separate handle for 6 rolls or error catching.
Also, `SendPlayAnimation`  takes a std::u16string& as it's second parameters which makes casting a bit messy.

Reply:
Casting to a u16string is not difficult all it takes is passing it through GeneralUtils::ASCIIToUTF16 (I think that was the name) and I strongly disagree that it is better to have a large switch case instead of a few lines of int -> utf16.

Reply:
```c++
int dieRoll = GeneralUtils::GenerateRandomNumber<int>(1, 6);
std::u16string anims[6] = {
    u"roll-die-1", u"roll-die-2", u"roll-die-3",
    u"roll-die-4", u"roll-die-5", u"roll-die-6",
};
GameMessages::SendPlayAnimation(self, anims[dieRoll - 1]);
if (dieRoll == 6) {
    // Mission code
}
```

Reply:
The above suggestion seems like a good middle ground, opinions @ckawell.

Reply:
Alright, thanks @Xiphoseer. I can add this later today. I'll ping you Jett when I have it committed.